exercises/practice/react/src/main/java/React.java

```diff
@@ -0,0 +1,44 @@
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class React<T> {
+
+    interface Cell<T> {
+        T getValue();
+
+        void addCallback(Consumer<T> consumer);
+
+        void removeCallback(Consumer<T> callback);
+    }
+
+    static class InputCell<T> implements Cell<T> {
```

Comment:
I'm not sure it makes sense to provide a stub class for input cells here, while not providing a stub class for compute cells. This might lead a student to get confused as for what to implement themselves.

Maybe it makes more sense to only add interfaces in the stub file, so that it's up to the student to derive concrete classes for them.

``` java
public class React {
    interface Cell<T> {
        T getValue();
    }

    interface InputCell<T> extends Cell<T> {
        void setValue(T newValue);
    }

    interface ComputeCell<T> extends Cell<T> {
        void addCallback(Consumer<T> callback);
        void removeCallback(Consumer<T> callback);
    }

    public static <T> InputCell<T> inputCell(T initialValue) {
        throw new UnsupportedOperationException("Please implement the React.inputCell() method");
    }

    public static <T> ComputeCell<T> computeCell(Function<List<T>, T> function, List<Cell<T>> cells) {
        throw new UnsupportedOperationException("Please implement the React.computeCell() method");
    }
}
```

Reply:
That sounds reasonable. I'll make the update.